src/main/java/ro/developmentfactory/thedrones/service/DronePositionServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+    private static final Logger log = LoggerFactory.getLogger(DronePositionServiceImpl.class);
+
+    public DronePositionServiceImpl(DroneStatusRepository droneStatusRepository, DroneRepository droneRepository) {
+        this.droneStatusRepository = droneStatusRepository;
+        this.droneRepository = droneRepository;
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse turnDroneLeft(DroneStatusResponse droneStatusResponse) {
+        log.debug("Turning drone left :{}",droneStatusResponse);
+        UUID idDrone = droneStatusResponse.getIdDrone();
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        DroneStatus currentStatus = drone.getDroneStatusList().getFirst();
+        currentStatus.setFacingDirection(currentStatus.getFacingDirection().turnLeft());
+
+        droneStatusRepository.save(currentStatus);
+       return convertToResponse(currentStatus);
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse turnDroneRight(DroneStatusResponse droneStatusResponse) {
+        log.debug("Turning drone right :{}",droneStatusResponse);
+        UUID idDrone = droneStatusResponse.getIdDrone();
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        DroneStatus currentStatus = drone.getDroneStatusList().getFirst();
+        currentStatus.setFacingDirection(currentStatus.getFacingDirection().turnRight());
+
+        droneStatusRepository.save(currentStatus);
+        return convertToResponse(currentStatus);
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse moveForward(DroneStatusResponse droneStatusResponse) {
+        log.debug("Moving forward for drone status :{}",droneStatusResponse);
+
+        DroneStatus droneStatus = droneStatusRepository.findById(droneStatusResponse.getIdDroneStatus())
+                .orElseThrow(() -> new EntityNotFoundException("Drone status with id " + droneStatusResponse.getIdDroneStatus() + " not found"));
+
+        int newX = droneStatus.getCurrentPositionX();
+        int newY = droneStatus.getCurrentPositionY();
+
+        switch (droneStatus.getFacingDirection()) {
+            case N:
+                newY = newY + 1;
+                break;
+            case S:
+                newY = newY - 1;
+                break;
+            case E:
+                newX = newX + 1;
+                break;
+            case W:
+                newX = newX - 1;
+                break;
+        }
+
+        if (newX < 0 || newX >= 10 || newY < 0 || newY >= 10) {
+            log.error("Drone cannot move because it’s at the edge of the field. Current position: ({}, {}), New position: ({}, {})", droneStatus.getCurrentPositionX(), droneStatus.getCurrentPositionY(), newX, newY);
```

Comment:
you have `500 Internal Server Error` with error:
```
{
    "timestamp": "2024-08-06T22:40:16.762879+02:00",
    "status": 500,
    "error": "Internal Server Error",
    "message": "An unexpected error occurred. Please try again later."
}
```
in case it moves where it should not. please check that a proper exception is returned